src/bp/ui-admin/src/Pages/Workspace/Bots/index.tsx

```diff
@@ -19,6 +19,7 @@ import React, { Component, Fragment } from 'react'
 import { connect } from 'react-redux'
 import { generatePath, RouteComponentProps } from 'react-router'
 import { Alert, Col, Row } from 'reactstrap'
+import { setupServerPackageLoop } from '~/telemetry/middleware'
```

Comment:
I don't really like the name middleware as first it seems like more than a middleware and 2nd "middleware" doesn't really mean anything if it's not joint with something else

Reply:
Would offlineServerMiddleware or something similar be better and more specific?

Reply:
yes much better